src/main/ml-modules/ext/com.marklogic.smart-mastering/matcher.xqy

```diff
@@ -135,7 +140,11 @@ declare function matcher:find-document-matches-by-options(
         else (),
         cts:or-query(
           $minimum-threshold-combinations
-        )
+        ),
+        let $blocks := matcher:get-blocks(fn:base-uri($document))
+        where $blocks
```

Comment:
will this always fn:exist()? seems like the code below is returning an array always.

Reply:
you're right. should be checking fn:exists($blocks/node())